extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/ArcThreadSetupAction.java

```diff
@@ -15,9 +15,7 @@ public ArcThreadSetupAction(ManagedContext managedContext) {
 
     @Override
     public ThreadState activateInitial() {
```

Comment:
This can be racy @Ladicek @mkouba ?

FYI the original code was performing a putLocal/getState not atomically; but even the change I made (that make it atomic) is not ensuring what happen to the old value replaced, if any

Reply:
The old code was never a problem - if it were we would have seen a ton of very weird RESTEasy Reactive + CDI issues

Reply:
@Ladicek got a test that show how to can break...
The fact that we have no users complaining is (probably) due to none that has done it yet or that the effects are not known

Reply:
I'd like to see that test

Reply:
mmm talking with Ladislav, I see his code was meant to make `RequestContext::getIfActive` to perform a racy initialization instead, while this one is a different case

Reply:
I do think something like `activateInitial()` is a good idea. I don't know enough about how/when RESTEasy Reactive calls this code to know whether this is actually racy. It seems quite possible that the request context that is being created here doesn't escape anywhere, so there's no problem here.

The test I have is for a different thing -- it can show that `RequestContext.getIfActive()` is incorrect. That's probably a different issue.

Reply:
I like the idea but I don't think that we need to add a new method but simply return `ContextState` from the `activate()` methods instead. I will try to push a new commit to this PR.

Reply:
In term of existing API, agree, it's not required; but having a method that assume some state (transition) can easily check if something isn't right. 
If we don't care about adding such validation, it's fine. I'm adding the proposed change, in case

Reply:
mmmm maybe the `activate` one (that's using `null` as parameter) could be still considered as the initial activation

Reply:
Both could be initial activations. It's semantically equivalent.

Reply:
I saw the commit and test fix, nice; so the check is already paying off